### PR TITLE
Feature/docker environment

### LINKED ISSUE #277: Create Docker Environment
This is a feature request to include a docker build of the policy server using Docker Compose. This can utilize a Postgres docker container, a Redis docker container, and the policy server as a docker container and automatically perform the tedious setup of connecting them all together. The user should still be able to pass in environment variables to change the features activated like in the original project. The docker environment would be for mostly testing purposes only.

### PR DESCRIPTION
**Pull request by crokita**
Fixes #277 

Depends on #275. Review should wait until merged. 

This PR is ready for review.

### Risk
This PR makes no API changes.

### Summary
Adds a new docker folder to the project, where everything can be run in a Docker container environment. Provided are instructions on how to run the environment and some environment variables to be conscious of. WebEngine support, Redis caching and certification generation are all possible.

For this PR review the line in `docker-compose.yml` that specifies the argument for what VERSION the policy server runs on should change to `develop` to bring in the recent required changes for the server to function in docker mode:

```
    build:
      context: .
      args:
        - VERSION=develop
```

### CODE REVIEW

**Review by renonick87 — CHANGES_REQUESTED** [review]
See feedback.

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2022, Livio, Inc.
+FROM node:12
+
+ARG VERSION=master
```
[the remark this answers is not in the document]
Comment (crokita) [reply docker/Dockerfile:4]: The docker-compose.yml file has priority in the VERSION that is set. It replaces the VERSION in the Dockerfile.

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    volumes:
+      - data:/usr/policy
+    env_file:
+      - .env
+    environment:
+      DB_HOST: "postgres"
+      DB_PASSWORD: "${DB_PASSWORD:-postgres}"
+      DB_USER: "${DB_USER:-postgres}"
+      DB_DATABASE: "${DB_DATABASE:-postgres}"
+      CACHE_HOST: "${CACHE_HOST:-redis}"
```
Comment (renonick87) [inline docker/docker-compose.yml:20]: Is there any reason not to use predefined values for the CACHE variables like with the DB_HOST? I would think that any other values wouldn't be able to connect to the redis instance.

Comment (crokita) [reply docker/docker-compose.yml]: Agreed. Hardcoded the redis values here

**Review by renonick87 — APPROVED** [review]

**Pull request merged**